packages/core/src/services/documents/runDocumentVersion.ts

```diff
@@ -1,32 +1,80 @@
-import { Chain, createChain } from '@latitude-data/compiler'
+import path from 'path'
+
+import {
+  Chain,
+  createChain,
+  readMetadata,
+  Document as RefDocument,
+} from '@latitude-data/compiler'
 import {
   ChainEvent,
   ChainEventTypes,
-  DocumentVersion,
+  Commit,
   LATITUDE_EVENT,
   PROVIDER_EVENT,
   ProviderApiKey,
 } from '$core/browser'
-import { findWorkspaceFromDocument } from '$core/data-access'
-import { Result } from '$core/lib'
+import { findWorkspaceFromCommit } from '$core/data-access'
+import { NotFoundError, Result } from '$core/lib'
 import { streamToGenerator } from '$core/lib/streamToGenerator'
-import { ProviderApiKeysRepository } from '$core/repositories'
+import {
+  DocumentVersionsRepository,
+  ProviderApiKeysRepository,
+} from '$core/repositories'
 import { z } from 'zod'
 
 import { ai } from '../ai'
 
 export async function runDocumentVersion({
-  document,
+  documentUuid,
+  commit,
   parameters,
 }: {
-  document: DocumentVersion
+  documentUuid: string
+  commit: Commit
```

Comment:
Nein keep the api as it was. The document version service receives the document version, and you can resolve all the rest from there. You don't need scopes it's not the service responsibility to ensure tenancy.

Reply:
You can't resolve the `commit`. I need to know which `commit` you're running the prompt on, because the `ref` tags have to resolve to documents in the same commit.

The `document` object can contain an older `commitId`, if that's when it was last modified, but if a referenced document has been edited after that commit, those changes would not appear if I just resolve the references using the documents from the `getDocumentsAtCommit(document.commitId)`

Reply:
buf ya